cpp/src/bitmask/null_mask.cu

```diff
@@ -128,8 +145,92 @@ CUDF_KERNEL void set_null_mask_kernel(bitmask_type* __restrict__ destination,
     }
   }
 }
+
+CUDF_KERNEL void set_null_mask_bulk_kernel(
+  cudf::device_span<bitmask_type* __restrict__> destinations,
```

Comment:
Do we know whether the `__restrict__` keyword actually helps speed anything up here? Is the compiler optimizing any code differently? It seems fairly specialized to throw in here.

Reply:
Yeah you are right, no need to add the `__restrict__` here as the only real effect, if any, may be in `set_null_mask_impl`